### ISSUE TITLE
Autocommand event for when a window is closed

### ISSUE THREAD

**Issue by blueyed** [vim/vim#742]
There is currently no way to reliably detect if a window has been closed.
A `WinClose` / `WinQuit` / `QuitPost` autocommand event would be useful.

Workarounds like the following won't handle the case when a non-current window gets closed for example:

``` vim
fun! MyHandleWinClose()
  if get(t:, '_win_count', 0) > winnr('$')
    doautocmd User MyAfterWinClose
  endif
  let t:_win_count = winnr('$')
endfun
augroup vimrc_user
  au!
  au BufWinEnter,WinEnter,BufDelete * call MyHandleWinClose()
augroup END
```

And even if this could be extended to work reliably, it's much easier to have a single autocommand for it.

Given that there is `QuitPre` already, `QuitPost` would fit - but on the other hand `WinClose` is a better fit with `WinLeave` and `WinEnter`.

I can also imagine a more general event like `VimLayoutChanged` (additionally), which would be triggered in case of layout changes in general (including windows being resized).

(My current use case is auto-resizing quickfix / location list windows with min/max values, according to the number of lines they contain - and that needs to get done when other windows get closed, too)

**Comment by matt1003** [comment]
+1

**Comment by obcat** [comment]
I guess this can be closed since `WinClosed` event is added in [8.2.3591](https://github.com/vim/vim/commit/23beefed73aadb243fb67cf944e3d60fe8c038bb).
